adr/0002-redesign-of-task-variables-and-or-inputs.md

```diff
@@ -0,0 +1,94 @@
+# 2. Redesign of Task Variables and/or Inputs
+
+Date: 2025-03-24
+
+## Status
+
+Accepted
+
+## Context
+
+Maru Runner currently supports two mechanisms for task configuration: Variables and Inputs. This duality has led to overlapping functionality, inconsistencies in user experience, and increased maintenance overhead. The competing solutions have created ambiguity for both new and existing users regarding which configuration method to use. The goal of this redesign is to simplify the system by removing redundant approaches, ensuring consistency, and reducing technical debt.
```

Comment:
To solidify my understanding, am I correct in reading this as "The main goal of this ADR is to simplify the UX Maru" and that "While it might be cumbersome, there is nothing 'missing' feature wise atm"?

Reply:
correct, nothing missing atm that we are hard blocked by.